### ISSUE TITLE
Remove dependency on DropwizardMetricsFactory in native-image's classpath dependencies when using Cassandra

### ISSUE THREAD

**Issue by adryen31** [quarkusio/quarkus#15522]
This problem appears only when I run a Quarkus' native runner.

Example, with a simple quarkus-app.jar I don't get the problem, but when I build a native image runner, this problem appears during each Cassandra requesting.

In my project I use following Quarkus' version:

```
<quarkus-plugin.version>1.12.1.Final</quarkus-plugin.version>
<quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
<quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
<quarkus.platform.version>1.12.1.Final</quarkus.platform.version>
```

And I built the application with "mvn clean package -Pnative -Dnative-image.xmx=9g" command line.It's very strange because, even if I set:

 "quarkus.cassandra.metrics.enable=false" without dependency to "java-driver-metrics-micrometer" (or micropofile or metrics-jmx" dependency)

Or  "quarkus.cassandra.metrics.enable=true" with dependency to "java-driver-metrics-micrometer" (or micropofile or metrics-jmx" dependency)

The bevaviour when the native image running, is the same ... ClassNotFoundException.I don't know with this problem appears only in native mode, but I search a mean to "disabled" the metrics' system (for more lightness, or even at worst have them at runtime, but without the error)

I tried with the most recent com.datastax.oss java-driver-code's version "4.10.0"

But I was hoping to have the behavior advertised in the documentation, but unless I’m wrong about being able to disable ... I just wish I didn’t have the problem

But for me it seems the DropwizardMetricsFactory class isn't registered for reflexion at the runtime. I know is possible to register it, but if I make this, should register all the linked classes located in ?
```
   <dependency>
            <groupId>com.datastax.oss</groupId>
            <artifactId>java-driver-core</artifactId>
            <version>4.10.0</version>
    </dependency>
```
I don't know the best to by default registe the Datastax classes which cause the problem, or fixe the problem by not searching to instanciate the factory is the application.propreties of metrics.enable is to "false" value.

I precise, I tried to use all following dependencies, but same thing, not found at the native's runtime:
<dependency>
            <groupId>com.datastax.oss</groupId>
            <artifactId>java-driver-metrics-micrometer</artifactId>
            <version>4.10.0</version>
        </dependency>
        <dependency>
            <groupId>io.dropwizard.metrics</groupId>
            <artifactId>metrics-jmx</artifactId>
            <version>4.1.18</version>
        </dependency>
        <dependency>
            <groupId>com.datastax.oss</groupId>
            <artifactId>java-driver-metrics-microprofile</artifactId>
            <version>4.10.0</version>
        </dependency>

**Comment by ebullient** [comment]
It is unclear from the above which metrics extension you're using (if you are using one at all). Can you confirm what Quarkus extensions you are using?

If you are using the quarkus micrometer extension, we should look at creating an extension in Quarkiverse for the datastax driver (to ensure the native behavior is handled properly). 

See here: https://github.com/quarkiverse/quarkus-micrometer-registry .. the pattern to follow to add support should be fairly self-explanatory.

**Comment by adryen31** [comment]
Hello @ebullient  thank you for your comment,
but, not at all. precisely I don’t want to use metrics, but even by disabling this, the runtime of the native image continues to want to instantiate this class, isn't normal.
by default, de metrics should be disabled, because I haven't imported/declared the dependency in my pom.xml, and I don't want to explicitly use the metrics ... but in order to force the "disabled" I tried to set "quarkus.cassandra.metrics.enabled=false" but not really working.
for summarized, I just want to run my native application without any problem with cassandra metrics because I don't use it, and I don't know why, the factory is requested/searched...
In attached files, my configuration of extensions
![metrics-stacktrace](https://user-images.githubusercontent.com/1407536/110248501-d7979680-7f71-11eb-9872-2040fcc4e0ac.png)
![documentation](https://user-images.githubusercontent.com/1407536/110248503-d8c8c380-7f71-11eb-8567-94d8ba2a5717.png)
![dependenecies](https://user-images.githubusercontent.com/1407536/110248504-d8c8c380-7f71-11eb-83f1-a76718aded1f.png)
![dependency](https://user-images.githubusercontent.com/1407536/110248505-d8c8c380-7f71-11eb-937a-f391aee505f2.png)
![datastax-dependencies](https://user-images.githubusercontent.com/1407536/110248506-d9615a00-7f71-11eb-83f0-9bb257d47b47.png)
![application-properties](https://user-images.githubusercontent.com/1407536/110248507-d9615a00-7f71-11eb-89c3-e4c3652758a1.png)

**Comment by ebullient** [comment]
ok. Your intention really was unclear. 

I'll have a look tomorrow, but it seems like the cassandra extension may need amendment to avoid dependency on metrics. It can be tricky to get right.

**Comment by adryen31** [comment]
Yes thank, and don't work on sunday :-)

Well, I never wanted to use that metric system. I would have liked to offer a solution, but I don’t know how to test, to propose a fixbug or even a workaround

**Comment by adryen31** [comment]
Hooooo !!! I found another interesting point/clue
Why the error of the ClassNotFound search a "com.datastax.oss.driver.internal.metrics.micrometer.DropwizardMetricsFactory" instead the really class packaged in the 4.10.0 datastax.driver.core with  "com.datastax.oss.driver.internal.**core**.metrics.micrometer.DropwizardMetricsFactory"  ?
Maybe the seeked class was moved to another packaged and the reference not update ...

![other-information2](https://user-images.githubusercontent.com/1407536/110255688-40dcd100-7f95-11eb-99a0-b264cf46817d.png)
![other-information](https://user-images.githubusercontent.com/1407536/110255689-41756780-7f95-11eb-9cf6-78cf6fb967a7.png)

**Comment by adryen31** [comment]
If anyone guide me, I can try to make a fix, if you want. But I'm noy really sure where to begin to trying to realize the solution. I canparticipate actively 😊

**Comment by adryen31** [comment]
Ok I checked in Datastax Driver Core directly, In fact the metric registry is instantiated by default like this:
![datastax-code](https://user-images.githubusercontent.com/1407536/110258153-5d7f0600-7fa1-11eb-8755-2f5da649e97c.png)

So I think, because the Datastax's Driver must have by default a metric, maybe In Quarkus, at least the "com.datastax.oss.driver.internal.core.metrics.DropwizardMetricsFactory" should be registered for reflexion by default, no ? What do you think ?

**Comment by aruffie** [comment]
I don't have the write on my ticket for status' changing ... but we can close it I think, linked to my reporting to Cassandra/Datastax:
https://datastax-oss.atlassian.net/browse/JAVA-2927

**Comment by ebullient** [comment]
Y. that's definitely better.

**Comment by gsmet** [comment]
Closing then, thanks!

**Comment by adutra** [comment]
Hi, for future reference, indeed this issue was fixed in the Cassandra driver 4.11.0, see [here](https://datastax-oss.atlassian.net/browse/JAVA-2927). Then the fix was incorporated in the Cassandra extension 1.0.1, see [here](https://github.com/datastax/cassandra-quarkus/issues/157).
